src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import core.basesyntax.enums.Type;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MIN = 1;
+    private static final int MAX = 20;
+    private static final int DEFAULT_RADIUS = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int indexOfType = random.nextInt(Type.values().length);
+        Type type = Type.values()[indexOfType];
+        String color = colorSupplier.getRandomColor();
+        int sideA = random.nextInt(MIN, MAX);
```

Comment:
create separate private method `randomSide()`  with random.nextInt(MIN, MAX); and use it for each type of figure